### PR TITLE
Applied Set to compare special VLANs collections

### LINKED ISSUE #506: False negative on Interface. all_tags_available
Hi,

```
kytos $> sw.interfaces[4].all_tags_available()
Out[10]: False

kytos $> sw.interfaces[4].available_tags
Out[11]: {'vlan': [[1, 4095]]}

kytos $> sw.interfaces[4].tag_ranges
Out[12]: {'vlan': [[1, 4095]]}

kytos $> sw.interfaces[4].special_available_tags
Out[13]: {'vlan': ['any', 'untagged']}

kytos $> sw.interfaces[4].special_tags
Out[14]: {'vlan': ['untagged', 'any']}

```

As you can see, only the order of untagged/any are wrong.

### PR DESCRIPTION
**Pull request by Alopalao**
Closes #506

### Summary

Applied a Set to all types of tags and compare available tags with restricted tag collection

### Local Tests
Replicated issue case

### End-to-End Tests
N/A

### CODE REVIEW

**Comment by Alopalao** [comment]
It can also be changed to `set()`, just for the comparison.

File: kytos/core/interface.py [hunk 1]
```diff
@@ -246,8 +246,12 @@ def all_tags_available(self) -> bool:
         """Return True if all tags are avaiable (no tags used),
          False otherwise"""
         with self._tag_lock:
-            return (self.available_tags == self.tag_ranges and
-                    self.special_available_tags == self.special_tags)
+            tags_bool = self.available_tags == self.tag_ranges
```
Comment (viniarck) [inline kytos/core/interface.py:249]: Aldo, I think you can keep less variable states if you were to leverage early returns. For instance:

```
if self.available_tags != self.tag_ranges:
    return False
```

And, similarly in the special_available_tags items loop, just return mid iteration if a difference is found. 

Finally, good point, a `set()` for the values of available_tags seems more suited and lightweight.

**Review by viniarck — COMMENTED** [review]
Great fix. Overall, looks good. Asked minor implementation detail changes though. 

Also, changelog needs to be updated.

File: kytos/core/interface.py (hunk 1 above)
Comment (Alopalao) [reply kytos/core/interface.py]: Applied early returns and changed from `Counter` to `set`.

**Review by viniarck — APPROVED** [review]
LGTM. Let's ship it @Alopalao, go ahead and merge it. 

Finally, let's not forget backporting this too since the issue has the backport labels.

**Pull request merged**